react-form-controls/src/newsletter-form.jsx

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+
+export default class NewsletterForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { email: '', timerID: null };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange(event) {
+    this.setState({ email: event.target.value });
+    this.logState();
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
```

Comment:
You just needed to console log the state here, this method is called once the form is submitted.